### PR TITLE
Fixes mozilla/experimenter#7585 docs(nimbus): Add rollouts documentation

### PR DESCRIPTION
**Pull request by eliserichards**
## Description (optional)

Adding basic documentation for rollouts. This uses https://mozilla-hub.atlassian.net/browse/EXP-1219 as a starting point.

todo: 
- [x] what is a rollout/how is a rollout different from an experiment
- [x] future work **(Question:** should this link to jira?)
- [x] formatting
- [x] **Question:** I slotted this under Docs -> Deep Dives -> Specifications.. is there a better place to put this? 
  - talked to Janet, and decided we'll add a new subheading under "deep dives"

## Issue that this pull request resolves (optional)

Closes: mozilla/experimenter#7585

image

### CODE REVIEW

File: docs/deep-dives/experimenter/rollouts.mdx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## When should I use a rollout instead of an experiment?
+
+When I have a “winning branch” from an in-flight experiment, I want to easily set some percentage of the non-experiment population to that feature configuration. To create a rollout from an experiment, the experiment is cloned and a new rollout is created. Through this clone, the population percentage can be changed. Additionally, a rollout can be manually created (without cloning) through the "Create new" button on the Nimbus home screen. To mark the new item as a rollout, check the "This is a rollout" box on the "Branches" page:
+<img title="Rollout checkbox" src="/img/deep-dives/rollout-checkbox.png" height="500"/>
+
+A rollout is different from an experiment in that the population size can be scaled up and down (even set all the way to zero) with no analytical repurcussions. Even if a rollout is set to have a population size of zero (essentially turning the feature "off"), the rollout may be edited again to increase the population size. Likewise, if the rollout is "fully rolled out" to 100% of the population, it may still be edited.
+> ⚠️  See "Future work", live editability
```
Comment (jaredlockhart) [inline docs/deep-dives/experimenter/rollouts.mdx:22]: Let's not even mention this until it's implemented.

File: docs/deep-dives/experimenter/rollouts.mdx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Rollouts
+
+> 💡  Note: this feature is still in progress. See "Future work".
+
+Rollouts are single-branch experiments that differ from a traditional experiment in a number of ways.
```
Comment (jaredlockhart) [inline docs/deep-dives/experimenter/rollouts.mdx:14]: It's not really right to say a rollout is an experiment, maybe just say a rollout differs froman experiment in a number of ways.

File: docs/deep-dives/experimenter/rollouts.mdx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+## Features
+
+### Interactions
+
+- When configuring the audience for a rollout of a given feature, I should see information about all live experiments/rollouts of the same feature so I clearly understand how that will impact my population sizing.
+  This can be done by filtering on the home screen by the given feature. All experiments/rollouts that have been created for that feature will appear in the filtered list.
+
+### Directory view
+
+- I should to view all live experiments and rollouts for a given feature.
```
Comment (jaredlockhart) [inline docs/deep-dives/experimenter/rollouts.mdx:33]: `I should to`?

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
This is a good start @eliserichards but I think it's missing some of the most impoirtant information about rollouts, ie the interaction between rollouts and experiments on a given client, which include
- A client can be enrolled in both a single experiment AND rollout for a given feature
- The experiment feature value takes precedence over the rollout feature value
- Rollouts use a separate bucketing namespace from experiments so you don't need to worry about the populations colliding

There's more relevant design information in the planning doc

https://docs.google.com/document/d/1jCIWgHfVVB_u5323atF8CqZgSfQ0b88RcAMYBvQZsH8/edit?usp=sharing

that we could incorporate into this.

I like the setup of thinking about it from a 'I want to...' user story kind of perspective, but we might want to have like two sections, one talking about 'this is what it does and how it works' and one like 'I want to X here's how you do it' type thing.

Also we'll also need to update this little bit over here:

https://github.com/mozilla/experimenter-docs/blob/main/docs/getting-started/desktop-feature-api.mdx#configuration-sources

to mention that it'll use the rollout value.

It's also a bit wonky having that explanation scoped inside the desktop feature API when it applies on both desktop and mobile, maybe we wanna pull that out somewhere more generic?  Or at least replicate it into the mobile docs?  Not sure.

But thank you for getting this started 🙏

**Comment by eliserichards** [comment]
I stole those three bullet points from your review @jaredlockhart. I didn't change the desktop docs, since I'm not sure if it should be there or if we should have a section within the rollouts docs that's linked from desktop & mobile? we can chat about this later

**Review by jaredlockhart — APPROVED** [review]
Yep let's get it going and fill in more as time goes on.  Thanks @eliserichards ! 🎉 🎉 🎉

**Pull request merged**